pallets/subtensor/src/lib.rs

```diff
@@ -1646,15 +1646,33 @@ pub mod pallet {
         OptionQuery,
     >;
     #[pallet::storage]
-    /// --- MAP (netuid, commit_epoch) --> VecDeque<(who, serialized_compressed_commit, reveal_round)> | Stores a queue of v3 commits for an account on a given netuid.
+    /// MAP (netuid, epoch) → VecDeque<(who, ciphertext, reveal_round)>
+    /// DEPRICATED for CRV3WeightCommitsV2
     pub type CRV3WeightCommits<T: Config> = StorageDoubleMap<
         _,
         Twox64Concat,
         NetUid,
         Twox64Concat,
-        u64,
+        u64, // epoch key
+        VecDeque<(
+            T::AccountId,
+            BoundedVec<u8, ConstU32<MAX_CRV3_COMMIT_SIZE_BYTES>>,
+            RoundNumber,
+        )>,
+        ValueQuery,
+    >;
+    #[pallet::storage]
+    /// MAP (netuid, epoch) → VecDeque<(who, commit_block, ciphertext, reveal_round)>
+    /// Stores a queue of v3 commits for an account on a given netuid.
+    pub type CRV3WeightCommitsV2<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        NetUid,
+        Twox64Concat,
+        u64, // epoch key
         VecDeque<(
```

Comment:
What do you think about the dedicated struct here? The struct comments could explain the meaning of these fields.